src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -988,6 +989,10 @@ public boolean isInteresting(GerritTriggeredEvent event) {
                                                     changeBasedEvent.getFiles(
                                                         new GerritQueryHandler(getServerConfig(event))))) {
                                 logger.trace("According to {} the event is interesting.", p);
+                                if (triggerParamRuntime == null) {
+                                        triggerParamRuntime = new GerritTriggerRuntimeParam();
```

Comment:
This is bad as the `GerritTrigger` object is one instance per __job__ so adding a field to it means adding a field to the job not the build.

Reply:
To be honest, I can't think of another way of doing it. Please give me some clues to which file is the right one for this purpose. 

As far as why I thought this was the *only* option: This is the only file that knows about list of files involved in a given Gerrit change(`changeBasedEvent.getFiles`), thus, able to check if the affected-files in a gerrit change match the filtered files of file-based-trigger configuration.

Reply:
You would need to carry the information over from somewhere around here to the build. data in builds are stored via actions, int this case the simplest would be to put the information into the `GerritCause` object that is added to the build when scheduled.
Also, here I don't know if you can be sure the event was interesting because of the file, but maybe :)